cheatsheets/Microservices_based_Security_Arch_Doc_Cheat_Sheet.md

```diff
@@ -0,0 +1,245 @@
+
+| Parameter name | Description | 
+| :--- | :--- |
+| Publisher service name (ID) | Specify publisher service name (ID) defined above
+| Subscriber service name (ID) | Specify subscriber service name (ID) defined above
+| Message queue (ID) | Specify message queue (ID) defined above
+| Short description | Shortly describe the purpose of communication (receiving of information or commands for a state-changing business function) and data passed between services (if possible, in therms of assets defined above)
+
+### Identify “asset-to-storage” relations
+
+Collect information on the parameters listed below related to each “asset-to-storage” relation.
+
+| Parameter name | Description | 
+| :--- | :--- |
+| Asset name (ID) | Asset name (ID) defined above
+| Storage name (ID) | Specify storage name (ID) defined above
+| Storage type | Specify storage type for the asset, e.g. "golden source" or "cache"
+
+## Create a graphical presentation of application architecture
+
+It is advisable to create graphical presentation of application architecture (building blocks and relations defined above) in form of services call graph or data flow diagram. In order to do that one can use special software tools (e.g. Enterprise Architect) or [DOT language](https://en.wikipedia.org/wiki/DOT_(graph_description_language)). See example of using DOT language [here](https://gist.github.com/vladgolubev/80c5523336ddec3859c0e90d9a070882).
+
+## Use collected information in secure software development practices
+
+Collected information may be useful for doing application security practices, e.g. during defining security requirements, threat modeling or security testing. Sections below contains examples of activities related to securing application architecture (as well as its mapping to OWASP projects) and tips for their implementation using information collected above.
+
+### Attack surface analysis
+
+#### Implementation tips
+
+To enumerate microservices endpoints that need to be tested during security testing and analyzed during threat modeling analyze data collected under the following sections:
+* "Identify and describe application-functionality services" (parameter “API definition”);
+* "Identify and describe infrastructure services" (parameter “Link to the service documentation”).
+
+#### Mapping to OWASP projects
+
+* [OWASP ASVS, V1 “Architecture, Design and Threat Modeling Requirements”, #1.1.2](https://github.com/OWASP/ASVS/blob/master/4.0/en/0x10-V1-Architecture.md#v1-architecture-design-and-threat-modeling-requirements)
+* [OWASP Attack Surface Analysis Cheat Sheet](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Attack_Surface_Analysis_Cheat_Sheet.md)
+
+### Data leakage analysis
+
+#### Implementation tips
+
+To analyze possible data leakage analyze data collected under the following sections: 
+* "Identify and describe data assets";

```

Comment:
Clean the bulleted points. Remove the `;` and the `" "` surrounding them

Reply:
Done